### DISCUSSION TITLE
Renovate is unable to correctly process a repo when current node version is different and `engine-strict` npm flag is enabled

### DISCUSSION THREAD

**Discussion by maxmil7** [renovatebot/renovate#9824] [Request Help]
**What Renovate type, platform and version are you using?**
Renovate v25.7.6
Installed renovate module locally and tried to renovate a repository

**Describe the bug**
Renovate opened a PR to update a dependency but reported this error in PR body:

```
⚠️ Artifact update problem
Renovate failed to update an artifact related to this branch. You probably do not want to merge this PR as-is.
```
```
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for xyznodeweb@0.1.0: wanted: {"node":"12"} (current: {"node":"14.16.1","npm":"6.14.12"})
npm ERR! notsup Not compatible with your version of node/npm: xyznodeweb@0.1.0
npm ERR! notsup Not compatible with your version of node/npm: xyznodeweb@0.1.0
npm ERR! notsup Required: {"node":"12"}
npm ERR! notsup Actual:   {"npm":"6.14.12","node":"14.16.1"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /var/folders/14/fz_fd9ds7fj_fjzrx5bwgkg838zp42/T/renovate
```

The app has this config in package.json
```
 "engines": {
    "node": "12"
  }
```

.npmrc
```
save-exact=true
engine-strict=true
```

**Have you created a minimal reproduction repository?**
If this indeed sounds like a bug, I can try to create a test repo to reproduce the issue and share a link.
But wanted to check first before working on it.

**Comment by maxmil7** [comment]
I removed the `engine-strict=true` setting from `.npmrc` file and renovate was able to process the repo and apply the `node 12` constraint.
But with the `engine-strict` flag enabled, it seems to be running into a problem.
Please let me know if you would like me to create a sample repo to demo this issue

**Comment by maxmil7** [comment]
Updated the issue title/description based on the above

**Comment by viceice** [comment]
You need to run renovate with `binarySource=docker`. Only in this mode renovate is able to switch native tools versions by using docker sidecar container.

If you are using the renovate docker image, you should switch to the slim image, as it is already prepared for docker mode and has no global tools, is it always uses docker sidecar container for it.

**Reply by geraintwhite** [reply]
Isn't `binarySource=install` also supposed to solve this? I did use `binarySource=docker` but ran into docker pull limits so switched to `binarySource=install` and now encounter the engines mismatch.

**Reply by viceice** [reply]
`install` isn't ready for dynamic node versions, so the docker sidecar are your only option beside customizing renovate docker image yourself.

**Reply by keeganwitt** [reply]
@viceice Is there an issue to track when `install` will support Node versions?

**Reply by rarkins** [reply]
It already does

**Reply by keeganwitt** [reply]
Sorry, I should have clarified. I right this comment was suggesting `install` would work for self-hosted. But I guess this was an older comment that predates this working on ContainerBase.

**Reply by rarkins** [reply]
If you are using the official `renovate/renovate` image (non-slim) then it now defaults to `binarySource=install` and will dynamically install `node` versions
